portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -62,9 +67,42 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(taskEntity);
  
-    response.sendRedirect("/index.html");
+    doGet(request,response);
```

Comment:
It is confusing to have the doPost method call the doGet method, which is supposed to handle incoming `GET` requests. Someone making changes to `doGet` later thinking that this is a top of the funnel method might not realize something else calls this method and might change the behavior of the `doPost` method by accident.

If you have common code that can be shared, you should create a separate private function and have `doGet` and `doPost` call that function.

Reply:
would i still need a doGet function? if i put everything inside of the Post method and comment out the doGet method it works the same. Or is it required to have a doGet method?

Reply:
In this case, since you are no longer using that route, then you don't need it anymore. Your javascript use to make a `GET` request to retrieve information from the java servelet, which is why it's needed. But in this PR, your javascript no longer makes a `GET` request, since it was updated to make a `POST` request.